servercom/servercom.go

```diff
@@ -0,0 +1,138 @@
+package serverCom
+
+import (
+	"fmt"
+	mm "github.com/mattermost/platform/model"
+)
+/*
+ServerCom acts as a mitigator between the frontend and the mattermost model API.
+*/
+type ServerCom struct {
+	Client mm.Client
+	Team mm.Team
+	Channel *mm.Channel
+}
+
+/*
+Startup accepts the url and login credentials for a user, and returns a new serverCom struct.
+*/
+func Startup(url string, username string, password string) ServerCom {
```

Comment:
A package running in parallel to the interface that just compared the length of the message list to a variable storing the previous length of the message length makes the most sense to me. If the length is greater then it updates the variable and returns non-nil.